### PR TITLE
gitui: Update to version 0.22.1

### PR DESCRIPTION
**Pull request by 0x1DA117**
#### Description

I stumbled upon this [issue](https://trac.macports.org/ticket/65654) and simply tried to update the port to see if this somehow fixes the error. Unfortunately it didn't.

Help is appreciated if someone has an idea. I will try to poke around openssl a bit.

This is the error I got.

```
info:build error: linking with `/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/compwrap/ld/usr/bin/clang` failed: exit status: 1
:info:build   |
:info:build   = note: "/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/compwrap/ld/usr/bin/clang" "-arch" "x86_64" "-m64" "/opt/local/var/macport
s/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/.tmp/rustcO1LOQ6/symbols.o" "/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA
117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/gitui-936cd804a6a519f7.gitui.b0a6d24e-cgu.0.rcgu.o" "-L" "/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_gi
thub.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps" "-L" "/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitu
i-0221_devel_gitui/gitui/work/gitui-0.22.1/target/release/deps" "-L" "/opt/local/lib" "-L" "/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui
-0.22.1/target/x86_64-apple-darwin/release/build/libssh2-sys-a8610e58543b436b/out/build" "-L" "/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gi
tui-0.22.1/target/x86_64-apple-darwin/release/build/openssl-sys-aa40186a5bdce093/out/openssl-build/install/lib" "-L" "/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_de
vel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/build/sys-info-56ed78ea8fab22cc/out" "-L" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib" "/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_githu
b.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/.tmp/rustcO1LOQ6/libsys_info-df5ac9c1b033c65b.rlib" "/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_dev
el_gitui/gitui/work/.tmp/rustcO1LOQ6/liblibssh2_sys-d55fea23fde17dcd.rlib" "/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/.tmp/rustcO1LOQ6/libo
penssl_sys-b066a644f6fda84d.rlib" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-21fbf052a8fe8ccb.rlib" "-framework" "CoreFoundation" "-framework" "CoreServices" "-framework" "CoreFoundation" "-lgit2" "-lz" "-liconv" "
-lSystem" "-lc" "-lm" "-L" "/opt/local/lib/rustlib/x86_64-apple-darwin/lib" "-o" "/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/ta
rget/x86_64-apple-darwin/release/deps/gitui-936cd804a6a519f7" "-Wl,-dead_strip" "-nodefaultlibs"
:info:build   = note: Undefined symbols for architecture x86_64:
:info:build             "_COMP_get_name", referenced from:
:info:build                 _do_load_builtin_compressions_ossl_ in libopenssl_sys-b066a644f6fda84d.rlib(ssl_ciph.o)
:info:build                (maybe you meant: _SSL_COMP_get_name)
:info:build             "_COMP_get_type", referenced from:
:info:build                 _do_load_builtin_compressions_ossl_ in libopenssl_sys-b066a644f6fda84d.rlib(ssl_ciph.o)
:info:build             "_COMP_zlib", referenced from:
:info:build                 _do_load_builtin_compressions_ossl_ in libopenssl_sys-b066a644f6fda84d.rlib(ssl_ciph.o)
:info:build             "_Camellia_cbc_encrypt", referenced from:
:info:build                 _camellia_init_key in libopenssl_sys-b066a644f6fda84d.rlib(e_camellia.o)
:info:build             "_Camellia_decrypt", referenced from:
:info:build                 _camellia_init_key in libopenssl_sys-b066a644f6fda84d.rlib(e_camellia.o)
:info:build             "_Camellia_encrypt", referenced from:
:info:build                 _camellia_init_key in libopenssl_sys-b066a644f6fda84d.rlib(e_camellia.o)
:info:build             "_Camellia_set_key", referenced from:
:info:build                 _camellia_init_key in libopenssl_sys-b066a644f6fda84d.rlib(e_camellia.o)
:info:build             "_ERR_load_COMP_strings", referenced from:
:info:build                 _err_load_crypto_strings_int in libopenssl_sys-b066a644f6fda84d.rlib(err_all.o)
:info:build             "_IDEA_cbc_encrypt", referenced from:
:info:build                 _idea_cbc_cipher in libopenssl_sys-b066a644f6fda84d.rlib(e_idea.o)
:info:build             "_IDEA_cfb64_encrypt", referenced from:
:info:build                 _idea_cfb64_cipher in libopenssl_sys-b066a644f6fda84d.rlib(e_idea.o)
:info:build             "_IDEA_ecb_encrypt", referenced from:
:info:build                 _idea_ecb_cipher in libopenssl_sys-b066a644f6fda84d.rlib(e_idea.o)
:info:build             "_IDEA_ofb64_encrypt", referenced from:
:info:build                 _idea_ofb_cipher in libopenssl_sys-b066a644f6fda84d.rlib(e_idea.o)
:info:build             "_IDEA_set_decrypt_key", referenced from:
:info:build                 _idea_init_key in libopenssl_sys-b066a644f6fda84d.rlib(e_idea.o)
:info:build             "_IDEA_set_encrypt_key", referenced from:
:info:build                 _idea_init_key in libopenssl_sys-b066a644f6fda84d.rlib(e_idea.o)
:info:build             "_SEED_cbc_encrypt", referenced from:
:info:build                 _seed_cbc_cipher in libopenssl_sys-b066a644f6fda84d.rlib(e_seed.o)
:info:build             "_SEED_cfb128_encrypt", referenced from:
:info:build                 _seed_cfb128_cipher in libopenssl_sys-b066a644f6fda84d.rlib(e_seed.o)
:info:build             "_SEED_ecb_encrypt", referenced from:
:info:build                 _seed_ecb_cipher in libopenssl_sys-b066a644f6fda84d.rlib(e_seed.o)
:info:build             "_SEED_ofb128_encrypt", referenced from:
:info:build                 _seed_ofb_cipher in libopenssl_sys-b066a644f6fda84d.rlib(e_seed.o)
:info:build             "_SEED_set_key", referenced from:
:info:build                 _seed_init_key in libopenssl_sys-b066a644f6fda84d.rlib(e_seed.o)
:info:build             "_comp_zlib_cleanup_int", referenced from:
:info:build                 _OPENSSL_cleanup in libopenssl_sys-b066a644f6fda84d.rlib(init.o)
:info:build           ld: symbol(s) not found for architecture x86_64
:info:build           clang: error: linker command failed with exit code 1 (use -v to see invocation)
:info:build           
:info:build error: could not compile `gitui` due to previous error
```

This is the commend that failed:

```
/opt/local/bin/rustc \
  --crate-name gitui \
  --edition=2021 src/main.rs \
  --error-format=json \
  --json=diagnostic-rendered-ansi,artifacts,future-incompat \
  --crate-type bin \
  --emit=dep-info,link \
  -C opt-l evel=z \
  -C lto \
  -C codegen-units=1 \
  --cfg 'feature="default"' \
  --cfg 'feature="gh-emoji"' \
  --cfg 'feature="ghemoji"' \
  --cfg 'feature="regex-fancy"' \
  --cfg 'feature="trace-libgit"' \
  --cfg 'feature="vendor-openssl"' \
  -C metadata=936cd804a6a519f7 \
  -C extra-filename=-936cd804a6a519f7 \
  --out-dir /opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps \
  --target x86_64-apple-darwin \
  -C linker=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/compwrap/ld/usr/bin/clang \
  -L dependency=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps \
  -L dependency=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/release/deps \
  --extern anyhow=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libanyhow-1cba9b32f7e16bd3.rlib \
  --extern asyncgit=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libasyncgit-46f8fbe4f7be5e04.rlib \
  --extern backtrace=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libbacktrace-957c5f7f273b62e9.rlib \
  --extern bitflags=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libbitflags-e41bf8979a091b8d.rlib \
  --extern bugreport=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libbugreport-d732c0f9e9794d6d.rlib \
  --extern bytesize=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libbytesize-17952a7fa268ca29.rlib \
  --extern chrono=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libchrono-8dae882b4bf9cdc9.rlib \
  --extern clap=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libclap-09bf54ce8880d495.rlib \
  --extern crossbeam_channel=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libcrossbeam_channel-cf4dae24d7ef31f9.rlib \
  --extern crossterm=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libcrossterm-e3b08ee4c2486f54.rlib \
  --extern dirs_next=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libdirs_next-49952951f1176b18.rlib \
  --extern easy_cast=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libeasy_cast-5134835a2d05217d.rlib \
  --extern filetreelist=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libfiletreelist-c5e56cc3a4c9aa83.rlib \
  --extern fuzzy_matcher=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libfuzzy_matcher-f329740362d93336.rlib \
  --extern gh_emoji=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libgh_emoji-d3f06e35c690fb6f.rlib \
  --extern itertools=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libitertools-41c1163548bcda19.rlib \
  --extern log=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/liblog-0c30846c895e8032.rlib \
  --extern notify=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libnotify-4e1c3c04e88b923a.rlib \
  --extern notify_debouncer_mini=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libnotify_debouncer_mini-eed50db800343175.rlib \
  --extern once_cell=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libonce_cell-721e1e872c6deee6.rlib \
  --extern rayon_core=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/librayon_core-ef42e4b83776a5a2.rlib \
  --extern ron=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libron-63fae2209eac71c8.rlib \
  --extern scopeguard=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libscopeguard-61e2e48ba0d2c889.rlib \
  --extern scopetime=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libscopetime-df10f10f6932ad0e.rlib \
  --extern serde=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libserde-491f33a12ae207fe.rlib \
  --extern simplelog=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libsimplelog-4f9a7c0b80c18d89.rlib \
  --extern syntect=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libsyntect-6131e8e12aa40260.rlib \
  --extern textwrap=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libtextwrap-cc512384eef2aefe.rlib \
  --extern tui=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libtui-01923d421239c44e.rlib \
  --extern unicode_segmentation=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libunicode_segmentation-1b5b1e52fcfaf986.rlib \
  --extern unicode_truncate=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libunicode_truncate-f942e066ca663e2e.rlib \
  --extern unicode_width=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libunicode_width-2ea4ffbc391a205d.rlib \
  --extern which=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/deps/libwhich-183de3e1e2b0680d.rlib \
  -L native=/opt/local/lib \
  -L native=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/build/libssh2-sys-a8610e58543b436b/out/build \
  -L native=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/build/openssl-sys-aa40186a5bdce093/out/openssl-build/install/lib \
  -L native=/opt/local/var/macports/build/_Users_danieltrautmann_Development_Projects_github.com_0x1DA117_macports-ports_gitui-0221_devel_gitui/gitui/work/gitui-0.22.1/target/x86_64-apple-darwin/release/build/sys-info-56ed78ea8fab22cc/out
```

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 13.2.1 22D68 x86_64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by 0x1DA117** [comment]
I added the `--features "vendor-openssl"` build argument, which fixes the issue. This pulls in OpenSSL as dependency of gitui (Cargo Crate), so that gitui gets its own OpenSSL.

I'm not sure if this is considered to be a good way of dealing with the issue, but it fixes it.

Let me know what you think.

File: devel/gitui/Portfile [hunk 1]
```diff
@@ -20,11 +19,11 @@ maintainers         {gmail.com:herby.gillot @herbygillot} \
...
 checksums           ${distname}${extract.suffix} \
-                    rmd160  c74a74b0f9c5bab7368d981fc99fb791b279db6a \
-                    sha256  eccec120dfd4c5c33a74d331a53f3d0c6cb7200e26b0c4c7873278eb951bc379 \
-                    size    22404575
+                    rmd160  e60a348a3e49c00e3e8dbea4d59fefd700b1feaa \
+                    sha256  285e86c136ee7f410fdd52c5284ccf0d19011cc5f7709e5e10bb02f439a218ea \
+                    size    24968678 \
 
-openssl.branch      1.1
+build.args          {--features "vendor-openssl"}
```
Comment (herbygillot) [inline devel/gitui/Portfile:26]: Lets change this to `build.args-append`

**Comment by herbygillot** [comment]
> I added the `--features "vendor-openssl"` build argument, which fixes the issue. This pulls in OpenSSL as dependency of gitui (Cargo Crate), so that gitui gets its own OpenSSL.
> 
> I'm not sure if this is considered to be a good way of dealing with the issue, but it fixes it.
> 
> Let me know what you think.

Thanks @0x1DA117, nice find, I think we'll go with it.

One minor comment and I think this should be good.

**Comment by 0x1DA117** [comment]
Done. Thanks for the feedback!

**Pull request merged**